Update the spec to remove the link to a particular spec version (#106)
eclipse/microprofile-telemetry: tracing/spec/src/main/asciidoc/microprofile-telemetry-tracing-spec.asciidoc

This specification defines the behaviors that allow MicroProfile applications to easily participate in an environment where distributed tracing is enabled via https://opentelemetry.io[OpenTelemetry] (a merger between https://opentracing.io[OpenTracing] and https://opencensus.io[OpenCensus]).

> The OpenTelemetry specification describes the cross-language requirements and expectations for all OpenTelemetry implementations. This specification is based on the https://github.com/open-telemetry/opentelemetry-java/tree/v{otel-java-version}[Java implementation v{otel-java-version}] of OpenTelemetry. An implementation of this MicroProfile Telemetry may consume a later patch release of the Java implemenation as long as the required TCKs pass successfully.

Reviewer: ```suggestion
The OpenTelemetry specification describes the cross-language requirements and expectations for all OpenTelemetry implementations. This specification is based on the https://github.com/open-telemetry/opentelemetry-java/tree/v{otel-java-version}[Java implementation v{otel-java-version}] of OpenTelemetry. An implementation of this MicroProfile Telemetry may consume a later patch release of the Java implementation as long as the required TCKs pass successfully.
```

Refer to the OpenTelemetry specification repo to understand some essential terms.

* https://opentelemetry.io/docs/specs/otel/overview/[OpenTelemetry Overview]

> * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md[Tracing API]

Reviewer: ```suggestion
* https://opentelemetry.io/docs/specs/otel/trace/api/[Tracing API]
```

I missed one :disappointed:

> Refer to the OpenTelemetry specification repo to understand some essential terms.
> 
> * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md[OpenTelemetry Overview]
> * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md[Tracing API]
> * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md[Baggage API]
> * https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/context[Context API]
> * https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md[Resource SDK]

Reviewer: ```suggestion
Refer to the OpenTelemetry specification repo to understand some essential terms.

* https://opentelemetry.io/docs/specs/otel/overview/[OpenTelemetry Overview]
* https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md[Tracing API]
* https://opentelemetry.io/docs/specs/otel/baggage/api/[Baggage API]
* https://opentelemetry.io/docs/specs/otel/context/[Context API]
* https://opentelemetry.io/docs/specs/otel/resource/sdk/[Resource SDK]
```

If we're linking to the latest version, we may as well link to the nicer version on the OTel website.

This specification defines the behaviors that allow MicroProfile applications to easily participate in an environment where distributed tracing is enabled via https://opentelemetry.io[OpenTelemetry] (a merger between https://opentracing.io[OpenTracing] and https://opencensus.io[OpenCensus]).

> The OpenTelemetry specification describes the cross-language requirements and expectations for all OpenTelemetry implementations. This specification is based on the https://github.com/open-telemetry/opentelemetry-java/tree/v{otel-java-version}[Java implementation v{otel-java-version}] of OpenTelemetry. An implementation of this MicroProfile Telemetry can use a later patch release of the Java implemenation.

Reviewer 2: I'd propose _binary compatible_ instead of _later patch_. We don't get many patch releases of otel-java, but the stable parts should remain compatible.

Author: @pdudits binary comaptible changes include minor updates, which should NOT be allowed. In the above sentence, I tried to clarify only later patch releases are allowed to be consumed not the other releases. This spec should lock down major.minor release but open micro releases.

Author: I have reworded a bit.

Refer to the OpenTelemetry specification repo to understand some essential terms.
* https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md[OpenTelemetry Overview]
* https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md[Tracing API]
* https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md[Baggage API]
* https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/context[Context API]
* https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md[Resource SDK]

> This specification pulls in the 

Reviewer 2: Unfinished sentence. I'd propose having a list of actual java packages/artifacts for which compatibility is expected.

[[sec:semantic-conventions]]
=== Semantic Conventions

> The https://github.com/open-telemetry/opentelemetry-java/tree/release/v{otel-java-version}/semconv/src/main/java/io/opentelemetry/semconv[Trace Semantic Conventions] for Spans and Attributes must be followed by any compatible implementation.

Reviewer: ```suggestion
The https://github.com/open-telemetry/opentelemetry-java/tree/v{otel-java-version}/semconv/src/main/java/io/opentelemetry/semconv[Trace Semantic Conventions] for Spans and Attributes must be followed by any compatible implementation.
```

URL is wrong so I've corrected it, but in addition, these files seem to have less information in them and less detail than the [main OTel spec](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.20.0/specification/trace/semantic_conventions).

In particular, it doesn't say which attributes are recommended, optional and required.

> // TODO The hard coded otel-spec-version need to be replaced in future spec releases!

Reviewer: Can remove this TODO now.

[IMPORTANT]
====

> This is a deviation from the OpenTelemetry Specification that specifies this configuration property officially, where https://opentelemetry.io[OpenTelemetry] is activated by default!

Reviewer: The autoconfigure docs also define the default value as `false` here: https://github.com/open-telemetry/opentelemetry-java/tree/v1.28.0/sdk-extensions/autoconfigure#disabling-opentelemetrysdk

Author: Dscussed with @Azquelt and we decided to leave it as it is.

[[sec:microprofile-telemetry-and-microprofile-opentracing]]
== MicroProfile Telemetry and MicroProfile OpenTracing

> If MicroProfile Telemetry and MicroProfile OpenTracing are both present in one application, it is advised only to enable one of them. Otherwise, no portable behaviour may occur.

Reviewer: ```suggestion
If MicroProfile Telemetry and MicroProfile OpenTracing are both present in one application, it is recommended to only enable one of them, otherwise non-portable behaviour may occur.
```
